manifests.go

```diff
@@ -0,0 +1,104 @@
+package distribution
+
+import (
+	"fmt"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+)
+
+// Manifest represents a registry object specifying a set of
+// references and an optional target
+type Manifest interface {
+	// Target returns a descriptor for the configuration of this manifest.  This
+	// may return a nil descriptor if a target is not applicable for the given manifest.
+	Target() Descriptor
```

Comment:
As I mentioned in https://github.com/docker/distribution/pull/1268/files#r47700977, I would prefer to remove this, because it only makes sense for schema2. It's analagous to `Signatures`, which only makes sense for schema1, and thus is not part of the `Manifest` interface.